cmd/incusd/main_forknet.go

```diff
@@ -71,6 +71,27 @@ static void forkdonetdetach(char *file) {
 	// Jump back to Go for the rest
 }
 
+static void forkdonetdhcp() {
+	char *pidstr;
+	char path[PATH_MAX];
+
+	pidstr = getenv("LXC_PID");
+	if (!pidstr) {
+		fprintf(stderr, "No LXC_PID in environment\n");
+		_exit(1);
+	}
+
+	snprintf(path, sizeof(path), "/proc/%s/ns/net", pidstr);
+
+	// Attach to the network namespace.
+	if (dosetns_file(path, "net") < 0) {
+		fprintf(stderr, "Failed setns to container network namespace: %s\n", strerror(errno));
+		_exit(1);
+	}
+
+	// Jump back to Go for the rest
+}
+
```

Comment:
Out of curiosity, why do we have C code here? As reference for the go implementation?

Reply:
Because Go can't safely do this. Any switch of network namespace as done here must be done from cgo prior to the Go runtime starting up.

Note that those aren't comments, it's actual code that gets compiled in and run, the comment syntax is just how cgo works.

Reply:
Oh, gotcha!